lib/src/debian.dart

```diff
@@ -0,0 +1,216 @@
+    () => fail("pkg.signingEmail must be set to deploy to PPA repository."));
+
+/// The data needed to generate the Control file in the Debian package.
+///
+/// This must be set explicitly.
+final controlData = InternalConfigVariable.fn<String>(() => fail(
+    "pkg.controlData must be set to generate Control file for the package."));
+
+/// Whether [addDebianTasks] has been called yet.
+var _addedDebianTasks = false;
+
+/// Adds the task to create and upload the new package to the PPA.
+void addDebianTasks() {
+  if (!Platform.isLinux) {
+    fail("Platform must be linux for this task.");
+  }
+
+  if (_addedDebianTasks) return;
+  _addedDebianTasks = true;
+
+  freezeSharedVariables();
+  debianRepo.freeze();
+  signingEmail.freeze();
+
+  addTask(GrinderTask('pkg-debian-update',
+      taskFunction: () => _update(),
+      description: 'Update the Debian package.'));
+}
+
+/// Releases the source code in a Debian package and
+/// updates the PPA repository with the new package.
+Future<void> _update() async {
+  final String packageName = standaloneName.value + "_" + version.toString();
+
+  var repo =
+      await cloneOrPull(url("https://github.com/$debianRepo.git").toString());
+
+  await _createDebianPackage(repo, packageName);
+  await _releaseNewPackage(repo);
+  await _gitUpdateAndPush(repo);
+}
+
+/// Creates a Debian package from the source code.
+Future<void> _createDebianPackage(String repo, String packageName) async {
+  String debianDir = await _createPackageDirectory(repo, packageName);
+
+  _generateControlFile(debianDir);
+  _generateExecutableFiles(debianDir);
+  // Pack the files into a .deb file
+  run("dpkg-deb", arguments: ["--build", packageName], workingDirectory: repo);
+  _removeDirectory(debianDir);
+}
+
+/// Delete the Directory with path [directory].
+Future<void> _removeDirectory(String directory) async {
+  var result = await Process.run("rm", ["-r", directory]);
+  if (result.exitCode != 0) {
+    fail('Unable to remove the directory\n${result.stderr}');
+  }
+}
+
+/// Scans the PPA [repo] for new packages and updates the
+/// release files, also signing them.
+Future<void> _releaseNewPackage(String repo) async {
+  await _updatePackagesFile(repo);
+  await _updateReleaseFile(repo);
+  await _updateReleaseGPGFile(repo);
+  await _updateInReleaseFile(repo);
+}
+
+/// Create the directory `repo/packageName` and relevant subfolders for the
+/// debian package.
+///
+/// Returns the path of created folder.
+Future<String> _createPackageDirectory(String repo, String packageName) async {
+  String debianDir = p.join(repo, packageName);
+  await Directory('$debianDir/DEBIAN').create(recursive: true);
+  await Directory('$debianDir/usr/local/bin').create(recursive: true);
+  return debianDir;
+}
+
+/// Generate the executable files from the map [executables] which contains the
+/// executable name as key and the path to the corresponding Dart file as value.
+void _generateExecutableFiles(String debianDir) {
```

Comment:
should this task regenerate the executable, or should it use one generated by other release steps @nex3 ?

Reply:
It should depend on an executable-generation task, similar to how the GitHub task does: https://github.com/google/dart_cli_pkg/blob/017d28b68fb9f4305216bfcd41d33ea6afdb708b/lib/src/github.dart#L260-L264

Reply:
well, for apt, it should depend on the linux generation task specifically.